src/ray/core_worker/core_worker.cc

```diff
@@ -5047,6 +5088,22 @@ void CoreWorker::UpdateTaskIsDebuggerPaused(const TaskID &task_id,
       worker::TaskStatusEvent::TaskStateUpdate(is_debugger_paused)));
 }
 
+void CoreWorker::HandleExecuteNcclSend(rpc::ExecuteNcclSendRequest request,
```

Comment:
Step 4: Sender actor uses this function to handle the `ExecuteNcclSend` RPC from the driver. Use `send_p2p_dependency_callback` which is Python callback to call `dist.send(tensor, dst_rank)`